pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -79,3 +79,79 @@ def append_argument(self):
         Get the argument which is passed to append().
         """
         return self._append_arg
+
+class ListInsert(PyccelInternalFunction):
+    """
+    Represents a call to the .insert() method.
+
+    Represents a call to the .insert() method of an object with a list type,
+    which inserts a given element at a given index in a list. This method returns `None`.
+    The insert method is called as follows:
+
+    >>> a = [2, 3, 4]
+    >>> a.insert(1, 1)
+    >>> print(a)
+    [1, 2, 3, 4]
+
+    Parameters
+    ----------
+    list_variable : Variable
+        The variable representing the list.
+
+    index : TypedAstNode
+        The index value for the element to be added.
+    
+    new_elem : Variable
+        The argument passed to insert() method.
```

Comment:
Why does the `new_elem` need to be a `Variable`? Is there a problem with:
```python
a : list[int] = []
a.insert(0, 1+2)
```
? Or does the semantic stage create a temporary variable here?

Reply:
That was my previous understanding (the semantic creates a temporary variable) but I was wrong, now that i've tested it with different values, `new_elem` could be a `Variable`, `Literal`, `PyccelAdd` ... in general, a `TypedAstNode`

Reply:
From what I remember the semantic creates a temporary variable when memory needs to be allocated. For simple objects like `1+2` this is not necessary so it just passes the object